### ISSUE TITLE
unicorn-emulate is broken

### ISSUE THREAD

**Issue by Grazfather** [hugsy/gef#312]
```python
======================================================================
FAIL: test_command_unicorn_emulate (__main__.TestGefCommands)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/runtests.py", line 406, in test_command_unicorn_emulate
    self.assertNoException(res)
  File "tests/runtests.py", line 33, in assertNoException
    raise AssertionError("Detected error in gdb output")
AssertionError: Detected error in gdb output

----------------------------------------------------------------------
Ran 1 test in 1.388s
```

This was detected after I fixed the tests: We were embarrassingly running some asserts that do nothing on failure.

**Comment by hugsy** [comment]
Can you elaborate on this "bug" ? I use this feature all the time and don't have any issue:

![](https://i.imgur.com/ByN0s4u.png)

**Comment by Grazfather** [comment]
Did some debugging: I transplanted the script and the dump that it generates onto another machine and it works fine. When running it on CI, however, it fails.

Turns out that this is simply because on CI the `python` binary is `python3`, when in your environment, it's `python2`.

The script we generate is not python3 compatible.

https://github.com/hugsy/gef/blob/67236fbc2f64cc5490aa5c43b3b2f14611ebecda/gef.py#L5206-L5209

The `str()` around the `code` is the problem.
